tests/pass-dep/shims/libc-fs.rs

```diff
@@ -160,7 +162,7 @@ fn test_rename() {
     remove_file(&path2).unwrap();
 }
 
-fn test_ftruncate() {
+fn test_ftruncate(ftruncate: unsafe extern "C" fn(fd: libc::c_int, length: i64) -> libc::c_int) {
```

Comment:
You need to make the `length` type generic; it is not the same between the two functions.

Reply:
I see, thanks!

Reply:
I might need more help on this. I am not too sure how can I make length generic. ``length`` can be either ``libc::off_t`` or ``libc::off64_t``. If I use:
```rust
fn test_ftruncate<T>(ftruncate: unsafe extern "C" fn(fd: libc::c_int, length: T) -> libc::c_int) {
```
then I will face problem casting ``i64`` to ``libc::off_t`` or ``libc::off64_t`` because the code below won't work. 
```rust
length:i64 = 5;
length_casted = 5 as T;
```
``Default`` used by ``tests/pass-dep/shims/mmap.rs`` is not applicable too as the value of length needs to be customised in the test.

Reply:
`T: From<i64>` or `T: TryFrom<i64>` should work